src/main/java/io/vertx/ext/couchdb/CouchdbClientOptions.java

```diff
@@ -0,0 +1,107 @@
+package io.vertx.ext.couchdb;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.codegen.json.annotations.JsonGen;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.WebClientOptions;
+
+@DataObject
+@JsonGen
+public class CouchdbClientOptions {
+
+  /**
+   * Default port for connecting with Couchdb server.
+   */
+  public static final int DEFAULT_PORT = 5984;
+
+  /**
+   * Default host for connecting with Couchdb server.
+   */
+  public static final String DEFAULT_HOST = "localhost";
+
+  private String host;
+  private int port;
+  private CouchdbCredentials credentials;
+  private String database;
+  private WebClientOptions webClientOptions;
+
+  public CouchdbClientOptions() {
+    this.host = DEFAULT_HOST;
+    this.port = DEFAULT_PORT;
+    this.webClientOptions = new WebClientOptions()
+      .setDefaultHost(DEFAULT_HOST)
+      .setDefaultPort(DEFAULT_PORT);
+  }
+
+  public CouchdbClientOptions(CouchdbClientOptions other) {
+    this.host = other.host;
+    this.port = other.port;
+    this.credentials = other.credentials;
+    this.database = other.database;
+    this.webClientOptions = other.webClientOptions != null ? other.webClientOptions : new WebClientOptions()
+      .setDefaultHost(other.host)
+      .setDefaultPort(other.port);
+  }
+
+  public CouchdbClientOptions(JsonObject json) {
+    this();
+    CouchdbClientOptionsConverter.fromJson(json, this);
+  }
+
+  public JsonObject toJson() {
+    JsonObject json = new JsonObject();
+    CouchdbClientOptionsConverter.toJson(this, json);
+    return json;
+  }
+
+  public String getHost() {
+    return host;
+  }
+
+  public CouchdbClientOptions setHost(String host) {
+    this.host = host;
+    if (this.webClientOptions != null) {
+      this.webClientOptions.setDefaultHost(host);
+    }
+    return this;
+  }
+
+  public int getPort() {
+    return port;
+  }
+
+  public CouchdbClientOptions setPort(int port) {
+    this.port = port;
+    if (this.webClientOptions != null) {
+      this.webClientOptions.setDefaultPort(port);
+    }
+    return this;
+  }
+
+  public CouchdbCredentials getCredentials() {
```

Comment:
Yes. The reason of why I added a new `CouchdbCredentials` was because I am using `@JsonGen` that can generate a `CouchdbClientOptionsConverter `class which can be used for converting from and to json like this:

```
public CouchdbClientOptions(JsonObject json) {
    this();
    CouchdbClientOptionsConverter.fromJson(json, this);
  }
```
However the `CouchdbClientOptionsConverter `class expects all the fields of the `CouchdbClientOptions` to be a concrete class thus I cannot use the `io.vertx.ext.auth.authentication.Credentials` class because it's an interface. But please let me know if you have any ideas around this or if we still want to generate a converter class in the first place.

Reply:
mark it as transient? Or redesign the options not to have credentials. It's important to stay with the interface here

Reply:
I would lean towards with redesigning the options. Maybe we can just remove the options class and simply use a builder pattern for the `CouchdbClient ` for simplicity and accessibility

Reply:
retire CouchDbClientOptions for webclient in constructor